robotoff/workers/tasks/import_image.py

```diff
@@ -33,59 +31,60 @@ def run_import_image_job(
-
+    image = get_image_from_url(image_url, error_raise=False, session=http_session)
 
-@with_db
-def import_image(barcode: str, image_url: str, ocr_url: str, server_domain: str):
-    product_store = get_product_store()
-    product = product_store[barcode]
-    image = save_image(barcode, image_url, product, server_domain)
+    if image is None:
+        return
 
-    if image is not None:
-        logger.info(f"New image {image.id} created in DB")
+    source_image = get_source_from_url(image_url)
 
-    predictions_all = get_predictions_from_image(barcode, image_url, ocr_url)
+    product = get_product_store()[barcode]
+    if product is None:
+        logger.warning(
+            f"Product {barcode} does not exist during image import ({source_image})"
+        )
+        return
 
-    for prediction_type, product_predictions in predictions_all.items():
-        if prediction_type == PredictionType.image_flag:
-            NotifierFactory.get_notifier().notify_image_flag(
-                product_predictions.predictions,
-                product_predictions.source_image,  # type: ignore
-                product_predictions.barcode,
+    with db:
+        with db.atomic():
+            save_image(barcode, source_image, product, server_domain)
+            import_insights_from_image(
+                barcode, image, source_image, ocr_url, server_domain
             )
-            continue
+        with db.atomic():
+            # Launch object detection in a new SQL transaction
+            run_object_detection(barcode, image, source_image, server_domain)
+
 
-    imported = import_insights(
-        predictions_all.values(),
-        server_domain,
-        automatic=True,
-        product_store=product_store,
+def import_insights_from_image(
+    barcode: str,
+    image: Image.Image,
+    source_image: str,
+    ocr_url: str,
+    server_domain: str,
+):
+    predictions_all = get_predictions_from_image(barcode, image, source_image, ocr_url)
+    NotifierFactory.get_notifier().notify_image_flag(
+        [p for p in predictions_all if p.type == PredictionType.image_flag],
+        source_image,
+        barcode,
     )
+    imported = import_insights(predictions_all, server_domain, automatic=True)
     logger.info(f"Import finished, {imported} insights imported")
 
 
 def save_image(
-    barcode: str, image_url: str, product: Optional[Product], server_domain: str
+    barcode: str, source_image: str, product: Product, server_domain: str
 ) -> Optional[ImageModel]:
     """Save imported image details in DB."""
-    if product is None:
-        logger.warning(
-            f"Product {barcode} does not exist during image import ({image_url})"
-        )
-        return None
-
-    source_image = get_source_from_image_url(image_url)
     image_id = pathlib.Path(source_image).stem
 
     if not image_id.isdigit():
-        logger.warning(f"Non raw image was sent: {image_url}")
+        logger.warning(f"Non raw image was sent: {source_image}")
```

Comment:
As you pass over code, it would be good to remove those usage of f-string while logging.

Recommended syntax is:
```suggestion
        logger.warning("Non raw image was sent: %s", source_image)
```

Note that I put formating arguments as args, I do not use % directly.

While usage of "%" formating seems cumbersome, it permits sentry to group errors using the string.

Reply:
Oh nice, I didn't know this. I'm fixing the log messages in save_image function, we can do the switch gradually on the rest of the codebase.